### PR TITLE
general info step: styles fixed and added & context connected & inputs validation added

### LINKED ISSUE #95: Implement layout design and setup for General Info step
Subtask of #79 

- Styles should be added
- Inputs should be validated
- Cities select should render only the cities of selected country 
- Form data should be stored in step-context

### PR DESCRIPTION
**Pull request by tufned**
### Subtask of #79 

- Styles added (1)
- Shared components styles fixed
- Inputs validation added (2)
- Cities select renders only cities of the selected country (3)
- Form data are connected and stored in `step-context`
- `GeneralInfoStep.js` changed to `.tsx`
- `/general-info-step/constants.ts` created 
- types for general info step added
- `GeneralInfoFormDataTransfer` function and `GeneralInfoFinal` interface added for later to send more pure data to backend on final step

---
### Updated
- `step-context` refactored 
      - `.js` -> `.ts`
      - `use-form` moved to context (so stepper popup uses only 1 `use-form` for all steps)
      - stepData type updated
- language and photo steps were refactored using updated `step-context`

---
## (1) Styles for mobile 
Знімок екрана 2024-08-04 о 11 52 49

## (2) Inputs validation added
Знімок екрана 2024-08-04 о 11 58 03

## (3) Cities select renders only cities of the selected country
Знімок екрана 2024-08-04 о 11 51 46
Знімок екрана 2024-08-04 о 11 52 05

### CODE REVIEW

**Review by HalynaProlubshchykova — APPROVED** [review]

**Review by SeredenkoDS — APPROVED** [review]
Wow, 20 files, that's a pretty decent chunk of work.
Looked briefly through the code, tested the styles and validation, tested that going back and forth between tabs no longer resets the data (which was brought up during demo).
Approving.

File: src/containers/tutor-home-page/general-info-step/GeneralInfoStep.tsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    dataRef.current = data
+  }, [data])
+
+  useEffect(() => {
+    if (generalInfo.data !== initialValues) handleDataChange(generalInfo.data)
+
+    return () => {
+      handleStepData('generalInfo', dataRef.current)
+    }
+  }, [])
```
Comment (vlad-khrychov) [inline src/containers/tutor-home-page/general-info-step/GeneralInfoStep.tsx]: Let`s discuss it at our meeting.

File: src/types/components/user-steps/generalInfoStep.types.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  professionalSummary: string
+}
+
+interface GeneralInfoFinal {
+  firstName: GeneralInfoForm['firstName']
+  lastName: GeneralInfoForm['lastName']
+  country: string | null
+  city: string | null
+  professionalSummary: string | null
+}
```
Comment (vlad-khrychov) [inline src/types/components/user-steps/generalInfoStep.types.ts]: Why we need 2 types?

File: src/types/components/user-steps/generalInfoStep.types.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  firstName: string
+  lastName: string
+  country: CountryType | null
+  city: CityType | null
+  professionalSummary: string
+}
+
+interface GeneralInfoFinal {
+  firstName: GeneralInfoForm['firstName']
+  lastName: GeneralInfoForm['lastName']
```
Comment (vlad-khrychov) [inline src/types/components/user-steps/generalInfoStep.types.ts]:
```suggestion
  firstName: GeneralInfoForm.firstName
  lastName: GeneralInfoForm.lastName

```

**Review by olehpona — APPROVED** [review]

**Review by SachkovaYuliia — APPROVED** [review]

**Review by MrAK-dev — APPROVED** [review]

File: src/hooks/use-steps.jsx [hunk 4]
```diff
@@ -62,23 +63,22 @@ const useSteps = ({ steps }) => {
...
+    //   firstName,
+    //   lastName,
+    //   address: {
+    //     country: country ?? '',
+    //     city: city ?? ''
+    //   },
+    //   professionalSummary: professionalSummary,
+    //   mainSubjects: stepData.subjects,
+    //   nativeLanguage: stepData.language ?? ''
+    // }
```
Comment (vlad-khrychov) [inline src/hooks/use-steps.jsx]: should we delete it?

**Review by vlad-khrychov — APPROVED** [review]

File: src/hooks/use-steps.jsx (hunk 4 above)
Comment (tufned) [reply src/hooks/use-steps.jsx]: yes, done)

**Pull request merged**